config/routes.rb

```diff
@@ -20,9 +20,13 @@
     resource :vote, controller: 'comments/votes', only: [:create, :destroy]
   end
 
-  get ':username' => 'users/lists#index', as: :user_profile
+  get ':reference_id/:id/delete' => 'links#destroy', as: :destroy_link
+  get ':username/:list_id/:id/update' => 'references#edit_abstract', as: :edit_abstract
+
```

Comment:
I forgot that it actually is an `edit` action, so it is a `get`. This should look like:

```ruby
delete ':reference_id/:id' => 'links#destroy', as: :destroy_link
get ':username/:list_id/:id/edit' => 'paper#edit'
```